backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/material/cockpit/QtyTUConvertor.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * #%L
+ * de.metas.ui.web.base
+ * %%
+ * Copyright (C) 2024 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.ui.web.material.cockpit;
+
+import de.metas.handlingunits.IHUCapacityBL;
+import de.metas.handlingunits.model.I_M_HU_PI_Item_Product;
+import de.metas.product.ProductId;
+import de.metas.quantity.Capacity;
+import de.metas.quantity.Quantity;
+import de.metas.uom.UomId;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+import org.compiere.model.I_C_UOM;
+
+import javax.annotation.Nullable;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.HashMap;
+import java.util.Map;
+
+@Value
+@Builder
+public class QtyTUConvertor implements QtyConvertor
+{
+	@NonNull IHUCapacityBL capacityBL;
+	@NonNull ProductId productId;
+	@NonNull I_M_HU_PI_Item_Product packingInstruction;
+	@NonNull I_C_UOM tuUOM;
+	@NonNull Map<UomId, Capacity> uomId2Capacity = new HashMap<>();
+
+	@Nullable
+	@Override
+	public Quantity convert(@Nullable final Quantity quantity)
+	{
+		if (quantity == null)
+		{
+			return null;
+		}
+
+		final Capacity capacity = getCapacityForUomId(quantity.getUOM());
+
+		final BigDecimal qtyTUQty = quantity.toBigDecimal()
+				.divide(capacity.toBigDecimal(), 2, RoundingMode.HALF_UP);
```

Comment:
do always up. event if a box is not filled, it's still one box

Reply:
hmmmmm, are we sure about this? I mean, you wouldn't be able to tell how much qty do you actually have

Reply:
we never do, that's what rounding is about.
but if you have your stuff in 5PCE-boxes and you have anything at all, then you have at least one box.

Reply:
alright, I thought this would be a bit different since they are checking the stock :-? but ok :D